pkg/scalers/splunk_observability_scaler.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/signalfx/signalflow-client-go/v2/signalflow"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+type splunkObservabilityMetadata struct {
+	TriggerIndex int
+
+	AccessToken           string  `keda:"name=accessToken,          order=authParams"`
+	Realm                 string  `keda:"name=realm,                order=authParams"`
+	Query                 string  `keda:"name=query,                order=triggerMetadata"`
+	Duration              int     `keda:"name=duration,             order=triggerMetadata"`
+	TargetValue           float64 `keda:"name=targetValue,   	   order=triggerMetadata"`
+	QueryAggregator       string  `keda:"name=queryAggregator,      order=triggerMetadata"`
+	ActivationTargetValue float64 `keda:"name=activationTargetValue, order=triggerMetadata"`
+}
+
+type splunkObservabilityScaler struct {
+	metadata  *splunkObservabilityMetadata
+	apiClient *signalflow.Client
+	logger    logr.Logger
+}
+
+func parseSplunkObservabilityMetadata(config *scalersconfig.ScalerConfig) (*splunkObservabilityMetadata, error) {
+	meta := &splunkObservabilityMetadata{}
+	meta.TriggerIndex = config.TriggerIndex
+
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, fmt.Errorf("error parsing splunk observability metadata: %w", err)
+	}
+
+	return meta, nil
+}
+
+func newSplunkO11yConnection(meta *splunkObservabilityMetadata, logger logr.Logger) (*signalflow.Client, error) {
+	if meta.Realm == "" || meta.AccessToken == "" {
+		return nil, fmt.Errorf("error: Could not find splunk access token or realm")
+	}
+
+	apiClient, err := signalflow.NewClient(
+		signalflow.StreamURLForRealm(meta.Realm),
+		signalflow.AccessToken(meta.AccessToken),
+		signalflow.OnError(func(err error) {
+			logger.Error(err, "error in SignalFlow client")
+		}))
+	if err != nil {
+		return nil, fmt.Errorf("error creating SignalFlow client: %w", err)
+	}
+
+	return apiClient, nil
+}
+
+func NewSplunkObservabilityScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	logger := InitializeLogger(config, "splunk_observability_scaler")
+
+	meta, err := parseSplunkObservabilityMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Splunk metadata: %w", err)
+	}
+
+	apiClient, err := newSplunkO11yConnection(meta, logger)
+	if err != nil {
+		return nil, fmt.Errorf("error establishing Splunk Observability Cloud connection: %w", err)
+	}
+
+	return &splunkObservabilityScaler{
+		metadata:  meta,
+		apiClient: apiClient,
+		logger:    logger,
+	}, nil
+}
+
+func (s *splunkObservabilityScaler) getQueryResult() (float64, error) {
+	comp, err := s.apiClient.Execute(context.Background(), &signalflow.ExecuteRequest{
+		Program: s.metadata.Query,
+	})
+	if err != nil {
+		return -1, fmt.Errorf("could not execute signalflow query: %w", err)
+	}
+
+	s.logger.Info("Started MTS stream.")
```

Comment:
Please keep only important stuff under INFO log level, this would be too verbose.

You can make all these `V(1)` or higher.


Bear in mind that these messages would be printed with each call to get metrics for all scaler instances, that could be a lot of noise

Reply:
Changed s.logger.Info to s.logger.V(1).Info